src/components/AugmentedResultList/AugmentedResultList.ts

```diff
@@ -0,0 +1,142 @@
+import { $$, ComponentOptions, IComponentBindings, Initialization, IQueryResult, IResultListOptions, IFieldOption } from 'coveo-search-ui';
+
+/**
+ * Interface for the object data returned from remote action.
+ *
+ * @export
+ * @interface IObjectData
+ */
+export interface IObjectData {
+    /**
+     * Data specific to a result with matching object id.
+     */
+    objectData: {}[];
+    /**
+     * Data to add to every result with matching object id.
+     */
+    commonData: {};
+}
+
+/**
+ * Generic interface for the response returned by the remote action method.
+ *
+ * @export
+ * @interface IPromiseReturnArgs
+ * @template T
+ */
+export interface IPromiseReturnArgs {
+    data: IObjectData;
+}
+
+export interface AugmentedResultListOptions extends IResultListOptions {
+    matchingIdField: IFieldOption;
+    objectDataAction: (objectIds: String[]) => Promise<IPromiseReturnArgs>;
+}
+
+export class AugmentedResultList extends Coveo.ResultList implements IComponentBindings {
+    static ID = 'AugmentedResultList';
+
+    /**
+     * @componentOptions
+     */
+    static options: AugmentedResultListOptions = {
+        /**
+         * The field to be used as matching ID between object and result.
+         */
+        matchingIdField: ComponentOptions.buildFieldOption({
+            required: true,
+        }),
+        /**
+         * The function used to fetch extra object information.
+         */
+        objectDataAction: ComponentOptions.buildCustomOption<(objectIds: String[]) => Promise<IPromiseReturnArgs>>(() => {
+            return null;
+        }),
+    };
+
+    /**
+     * Creates a new `AugmentedResultList` component.
+     * @param element The HTMLElement on which to instantiate the component.
+     * @param options The options for the `ResultList` component.
+     * @param bindings The bindings that the component requires to function normally. If not set, these will be
+     * automatically resolved (with a slower execution time).
+     */
+    constructor(public element: HTMLElement, public options: AugmentedResultListOptions, public bindings: IComponentBindings) {
+        super(element, ComponentOptions.initComponentOptions(element, AugmentedResultList, options), bindings, AugmentedResultList.ID);
+    }
+
+    private getObjectPayload(results: IQueryResult[]): String[] {
+        const field = this.getFieldString(this.options.matchingIdField);
+        if (results.length > 0) {
+            return results.filter((result) => result.raw && result.raw[field]).map((result) => result.raw[field]);
+        }
+        return [];
+    }
+
+    private getFieldString(fieldName: IFieldOption) {
+        return fieldName.replace('@', '');
+    }
+
+    protected enableAnimation() {
+        if (!document.getElementById('overlay')) {
+            $$(document.body).append($$('div', { id: 'overlay', class: 'modal-backdrop fade in' }).el);
```

Comment:
This is really odd, and I have doubt that it'll work well in Salesforce because we cannot 'escape' our components.

Could you explain what this does, so we can find an alternative?
(Also, I would avoid using `$$` when you can use native methods ;) )

Reply:
Iirc that's from the original result list

Reply:
I checked the code of ResultList.ts, I don't think so. @nathanlb, I'd double-check this while reworking on the PR if I were you ;)

Reply:
I thought it was too. but it isn't so i removed it. Must have to do with the ecolab custo. The display behaviour is fine without it